perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2677,6 +2673,24 @@ public List<ApplicationFormItemWithPrefilledValue> getFormItemsWithPrefilledValu
 
 			}
 
+			// prefill USERNAME items with reserved logins
+			if (itemW.getFormItem().getType() == USERNAME || itemW.getFormItem().getType() == PASSWORD) {
+				String destAttribute = itemW.getFormItem().getPerunDestinationAttribute();
```

Comment:
I'm undecided if we should use source or destination attribute here. Generally when pre-filling form items we check source attribute. We could theoretically want to pre-fill login from another namespace (some kind of migration or requirement for the same login), but in the end we store reserved logins by the destination attribute namespace, so I think its fine like this.